drivers/periph_common/cpuid.c

```diff
@@ -31,6 +31,11 @@
 #ifdef CPUID_ADDR
 void cpuid_get(void *id)
 {
+/* gcc builtin bounds checking raises the following false positive warnings */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"
+#pragma GCC diagnostic ignored "-Wstringop-overread"
```

Comment:
CI does not like it, but does this work?

```C
static inline void _copy(void *dst, size_t len, uint8_t src[len])
{
    memcpy(dst, src, len);
}

void cpuid_get(void *id)
{
    _copy(id, CPUID_LEN, (void *)CPUID_ADDR);
}
```

Reply:
```
In function '_copy',
    inlined from 'cpuid_get' at /home/fabian/forks/RIOT/drivers/periph_common/cpuid.c:39:5:
/home/fabian/forks/RIOT/drivers/periph_common/cpuid.c:33:5: error: 'memcpy' offset [0, 7] is out of the bounds [0, 0] [-Werror=array-bounds]
   33 |     memcpy(dst, src, len);
      |     ^~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

same issue

Reply:
Same with

```C
static inline void _copy(size_t len, uint8_t dst[len], uint8_t src[len])
{
    memcpy(dst, src, len);
}
```
?

Reply:
yes :/